### PR TITLE
Docs: Add docs for variable interpolation in plugins

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

Adds a guide on how to interpolate template variables for panel and data source plugins.

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]

File: docs/sources/developers/plugins/variables.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
++++
+title = "Interpolate variables in plugins"
+type = "docs"
++++
+
+# Interpolate variables in plugins
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/variables.md:6]: Please update the file name to match the topic title.

File: docs/sources/developers/plugins/variables.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Variables are placeholders for values, and can be used to create things like templated queries. For more information on variables, refer to [Templates and variables]({{< relref "../../variables/templates-and-variables.md" >}}).
+
+This guide explains how to leverage template variables in your panel plugins and data source plugins.
+
+We'll see how you can turn a string like this:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/variables.md:12]: OMG, this is a GREAT thing to have in an intro.

File: docs/sources/developers/plugins/variables.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
++++
+title = "Interpolate variables in plugins"
+type = "docs"
++++
+
+# Interpolate variables in plugins
+
+Variables are placeholders for values, and can be used to create things like templated queries. For more information on variables, refer to [Templates and variables]({{< relref "../../variables/templates-and-variables.md" >}}).
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/variables.md:8]:
```suggestion
Variables are placeholders for values and can be used to create things like templated queries and dashboard or panel links. For more information on variables, refer to [Templates and variables]({{< relref "../../variables/templates-and-variables.md" >}}).
```

File: docs/sources/developers/plugins/variables.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+into
+
+```sql
+SELECT * FROM services WHERE id = "auth-api"
+```
+
+Grafana provides a couple of helper functions to interpolate variables in a string template. Let's see how you can use them in your plugin.
+
+## Panels
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/variables.md:26]: This heading and the next both seem very vague. Can you make them more specific? WHAT about panels? Support panel variables? Add support for variables in panels to your plugin? I really don't know what this section is about based on the headings.

File: docs/sources/developers/plugins/variables.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+For panels, the [replaceVariables]({{< relref "../../packages_api/data/panelprops.md#replacevariables-property" >}}) function is available in the [PanelProps]({{< relref "../../packages_api/data/panelprops.md" >}}).
+
+Add `replaceVariables` to the argument list, and pass it a user-defined template string:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/variables.md:30]:
```suggestion
Add `replaceVariables` to the argument list, and pass it a user-defined template string.
```

Comment (oddlittlebird) [reply docs/sources/developers/plugins/variables.md:30]: As I commented in your other PR, you can go either way with the colons as long as you are consistent in your usage.

File: docs/sources/menu.yaml [hunk 6]
```diff
@@ -345,6 +345,8 @@
           link: /developers/plugins/metadata/
         - name: Data frames
           link: /developers/plugins/data-frames/
+        - name: Variables
```
Comment (oddlittlebird) [inline docs/sources/menu.yaml:348]: Please make this side menu label more descriptive. Plugin variables? Variable support?

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
More great plugin content!

**Review by oddlittlebird — APPROVED** [review]

**Pull request merged**